faith_presentation/src/commonMain/kotlin/net/thechance/mena/faith/presentation/feature/quran/search/SearchScreen.kt

```diff
@@ -0,0 +1,262 @@
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import mena.faith_presentation.generated.resources.Res
+import mena.faith_presentation.generated.resources.aya
+import mena.faith_presentation.generated.resources.ic_arrow
+import mena.faith_presentation.generated.resources.ic_clear
+import mena.faith_presentation.generated.resources.ic_outline_search
+import mena.faith_presentation.generated.resources.ic_search
+import mena.faith_presentation.generated.resources.ic_search_warning
+import mena.faith_presentation.generated.resources.no_results_found_subtitle
+import mena.faith_presentation.generated.resources.no_results_found_title
+import mena.faith_presentation.generated.resources.shadow
+import mena.faith_presentation.generated.resources.start_searching_subtitle
+import mena.faith_presentation.generated.resources.start_searching_title
+import net.thechance.mena.designsystem.presentation.component.text.Text
+import net.thechance.mena.designsystem.presentation.component.textField.TextField
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import net.thechance.mena.faith.presentation.base.ObserveAsEffect
+import net.thechance.mena.faith.presentation.feature.quran.surah.component.getAyahTextStyle
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+import org.koin.compose.viewmodel.koinViewModel
+import org.koin.core.parameter.parametersOf
+
+@Composable
+fun SearchScreen(
+    surahId: Int?,
+    surahName: String?,
+    viewModel: SearchViewModel = koinViewModel(parameters = { parametersOf(surahId, surahName) })
+) {
+    val state by viewModel.uiState.collectAsStateWithLifecycle()
+    ObserveAsEffect(viewModel.uiEffect) {
+        when (it) {
+            SearchEffect.NavigateBack -> {}
+            is SearchEffect.NavigateToSurah -> {}
+        }
+    }
+    Content(state, viewModel)
+}
+
+@Composable
+private fun Content(
+    state: SearchScreenState,
+    interactionListener: SearchInteractionListener
```

Comment:
you can just rename it `listener`
```suggestion
    listener: SearchInteractionListener
```